Seems like the "Wait for CI images" step failure could be related?
```
ERROR tests/system/providers/amazon/aws/example_ec2.py - botocore.exceptions.NoCredentialsError: Unable to locate credentials
```
```
______ ERROR collecting tests/system/providers/amazon/aws/example_ec2.py _______
tests/system/providers/amazon/aws/example_ec2.py:91: in <module>
    image_id = _get_latest_ami_id()
tests/system/providers/amazon/aws/example_ec2.py:52: in _get_latest_ami_id
    Filters=[{"Name": "description", "Values": [image_prefix]}], Owners=["amazon"]
/usr/local/lib/python3.7/site-packages/botocore/client.py:530: in _api_call
    return self._make_api_call(operation_name, kwargs)
/usr/local/lib/python3.7/site-packages/botocore/client.py:944: in _make_api_call
    operation_model, request_dict, request_context
/usr/local/lib/python3.7/site-packages/botocore/client.py:966: in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:119: in make_request
    return self._send_request(request_dict, operation_model)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:198: in _send_request
    request = self.create_request(request_dict, operation_model)
/usr/local/lib/python3.7/site-packages/botocore/endpoint.py:137: in create_request
    operation_name=operation_model.name,
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:412: in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:256: in emit
    return self._emit(event_name, kwargs)
/usr/local/lib/python3.7/site-packages/botocore/hooks.py:239: in _emit
    response = handler(**kwargs)
/usr/local/lib/python3.7/site-packages/botocore/signers.py:105: in handler
    return self.sign(operation_name, request)
/usr/local/lib/python3.7/site-packages/botocore/signers.py:189: in sign
    auth.add_auth(request)
/usr/local/lib/python3.7/site-packages/botocore/auth.py:418: in add_auth
    raise NoCredentialsError()
E   botocore.exceptions.NoCredentialsError: Unable to locate credentials
```

--- a/airflow/providers/amazon/aws/operators/ec2.py
+++ b/airflow/providers/amazon/aws/operators/ec2.py
@@ -218,11 +218,11 @@ class EC2TerminateInstanceOperator(BaseOperator):
         in the `terminated` state before returning.
     """
 
-    template_fields: Sequence[str] = ("instance_id", "region_name", "aws_conn_id", "wait_for_completion")
+    template_fields: Sequence[str] = ("instance_ids", "region_name", "aws_conn_id", "wait_for_completion")
 
     def __init__(
         self,
-        instance_id: str | list[str],
+        instance_ids: str | list[str],
         aws_conn_id: str = "aws_default",
         region_name: str | None = None,
         poll_interval: int = 20,
@@ -231,7 +231,7 @@ class EC2TerminateInstanceOperator(BaseOperator):
         **kwargs,
     ):
         super().__init__(**kwargs)
-        self.instance_ids = [*instance_id]
+        self.instance_ids = instance_ids
         self.aws_conn_id = aws_conn_id
         self.region_name = region_name
         self.poll_interval = poll_interval
@@ -239,6 +239,8 @@ class EC2TerminateInstanceOperator(BaseOperator):
         self.wait_for_completion = wait_for_completion
 
     def execute(self, context: Context):
+        if isinstance(self.instance_ids, str):
+            self.instance_ids = [self.instance_ids]
         ec2_hook = EC2Hook(aws_conn_id=self.aws_conn_id, region_name=self.region_name, api_type="client_type")
         ec2_hook.conn.terminate_instances(InstanceIds=self.instance_ids)
 

--- a/airflow/providers/amazon/aws/operators/ec2.py
+++ b/airflow/providers/amazon/aws/operators/ec2.py
@@ -116,3 +116,116 @@ class EC2StopInstanceOperator(BaseOperator):
             target_state="stopped",
             check_interval=self.check_interval,
         )
+
+
+class EC2CreateInstanceOperator(BaseOperator):
+    """
+    Create and start an EC2 Instance using boto3
+
+    :param image_id: ID of the AMI used to create the instance.
+    :param max_count: Maximum number of instances to launch. Defaults to 1.
+    :param min_count: Minimum number of instances to launch. Defaults to 1.
+    :param aws_conn_id: AWS connection to use
+    :param region_name: AWS region name associated with the client.
+    :param poll_interval: Number of seconds to wait before attempting to
+        check state of instance. Only used if wait_for_completion is True. Default is 20.
+    :param max_attempts: Maximum number of attempts when checking state of instance.
+        Only used if wait_for_completion is True. Default is 20.
+    :param config: Dictionary for arbitrary parameters to the boto3 run_instances call.
+    :param wait_for_completion: If True, the operator will wait for the instance to be
+        in the `running` state before returning.
+    """
+
+    template_fields: Sequence[str] = ("image_id", "region_name", "config")
+
+    def __init__(
+        self,
+        image_id: str,
+        max_count: int = 1,
+        min_count: int = 1,
+        aws_conn_id: str = "aws_default",
+        region_name: str | None = None,
+        poll_interval: int = 20,
+        max_attempts: int = 20,
+        config: dict | None = None,
+        wait_for_completion: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.image_id = image_id
+        self.max_count = max_count
+        self.min_count = min_count
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.poll_interval = poll_interval
+        self.max_attempts = max_attempts
+        self.config = config or {}
+        self.wait_for_completion = wait_for_completion
+
+    def execute(self, context: Context):
+        ec2_hook = EC2Hook(aws_conn_id=self.aws_conn_id, region_name=self.region_name, api_type="client_type")
+        instance_id = ec2_hook.conn.run_instances(
+            ImageId=self.image_id,
+            MinCount=self.min_count,
+            MaxCount=self.max_count,
+            **self.config,
+        )["Instances"][0]["InstanceId"]
+        self.log.info("Created EC2 instance %s", instance_id)
+        if self.wait_for_completion:
+            ec2_hook.get_waiter("instance_running").wait(
+                InstanceIds=[instance_id],
+                WaiterConfig={
+                    "Delay": self.poll_interval,
+                    "MaxAttempts": self.max_attempts,
+                },
+            )
+
+        return instance_id
+
+
+class EC2TerminateInstanceOperator(BaseOperator):
+    """
+    Terminate an EC2 Instance using boto3
+
+    :instance_id: ID of the instance to be terminated.
+    :param aws_conn_id: AWS connection to use
+    :param region_name: AWS region name associated with the client.
+    :param poll_interval: Number of seconds to wait before attempting to
+        check state of instance. Only used if wait_for_completion is True. Default is 20.
+    :param max_attempts: Maximum number of attempts when checking state of instance.
+        Only used if wait_for_completion is True. Default is 20.
+    """
+
+    template_fields: Sequence[str] = ("instance_id", "region_name")
+
+    def __init__(
+        self,
+        instance_id: str,
+        aws_conn_id: str = "aws_default",
+        region_name: str | None = None,
+        poll_interval: int = 20,
+        max_attempts: int = 20,
+        wait_for_completion: bool = False,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.instance_id = instance_id
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.poll_interval = poll_interval
+        self.max_attempts = max_attempts
+        self.wait_for_completion = wait_for_completion
+
+    def execute(self, context: Context):
+        ec2_hook = EC2Hook(aws_conn_id=self.aws_conn_id, region_name=self.region_name, api_type="client_type")
+        ec2_hook.conn.terminate_instances(InstanceIds=[self.instance_id])
+
+        self.log.info("Terminating EC2 instance %s", self.instance_id)
+        if self.wait_for_completion:
+            ec2_hook.get_waiter("instance_terminated").wait(
+                InstanceIds=[self.instance_id],
+                WaiterConfig={
+                    "Delay": self.poll_interval,
+                    "MaxAttempts": self.max_attempts,
+                },
+            )

--- a/airflow/providers/amazon/aws/operators/ec2.py
+++ b/airflow/providers/amazon/aws/operators/ec2.py
@@ -207,7 +207,7 @@ class EC2TerminateInstanceOperator(BaseOperator):
         For more information on how to use this operator, take a look at the guide:
         :ref:`howto/operator:EC2TerminateInstanceOperator`
 
-    :instance_id: ID of the instance to be terminated.
+    :param instance_id: ID of the instance to be terminated.
     :param aws_conn_id: AWS connection to use
     :param region_name: AWS region name associated with the client.
     :param poll_interval: Number of seconds to wait before attempting to

--- a/docs/apache-airflow-providers-amazon/operators/ec2.rst
+++ b/docs/apache-airflow-providers-amazon/operators/ec2.rst
@@ -85,6 +85,7 @@ To terminate an Amazon EC2 instance you can use
     :dedent: 4
     :start-after: [START howto_operator_ec2_terminate_instance]
     :end-before: [END howto_operator_ec2_terminate_instance]
+
 Sensors
 -------
 

--- a/docs/apache-airflow-providers-amazon/operators/ec2.rst
+++ b/docs/apache-airflow-providers-amazon/operators/ec2.rst
@@ -58,6 +58,31 @@ To stop an Amazon EC2 instance you can use
     :start-after: [START howto_operator_ec2_stop_instance]
     :end-before: [END howto_operator_ec2_stop_instance]
 
+
+Create and start an Amazon EC2 instance
+=======================================
+
+To create and start an Amazon EC2 instance you can use
+:class:`~airflow.providers.amazon.aws.operators.ec2.EC2CreateInstanceOperator`.
+
+.. exampleinclude:: /../../tests/system/providers/amazon/aws/example_ec2.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_ec2_create_instance]
+    :end-before: [END howto_operator_ec2_create_instance]
+
+
+Terminate an Amazon EC2 instance
+================================
+
+To terminate an Amazon EC2 instance you can use
+:class:`~airflow.providers.amazon.aws.operators.ec2.EC2TerminateInstanceOperator`.
+
+.. exampleinclude:: /../../tests/system/providers/amazon/aws/example_ec2.py
+    :language: python
+    :dedent: 4
+    :start-after: [START howto_operator_ec2_terminate_instance]
+    :end-before: [END howto_operator_ec2_terminate_instance]
 Sensors
 -------
 

--- a/tests/providers/amazon/aws/operators/test_ec2.py
+++ b/tests/providers/amazon/aws/operators/test_ec2.py
@@ -20,23 +20,83 @@ from __future__ import annotations
 from moto import mock_ec2
 
 from airflow.providers.amazon.aws.hooks.ec2 import EC2Hook
-from airflow.providers.amazon.aws.operators.ec2 import EC2StartInstanceOperator, EC2StopInstanceOperator
+from airflow.providers.amazon.aws.operators.ec2 import (
+    EC2CreateInstanceOperator,
+    EC2StartInstanceOperator,
+    EC2StopInstanceOperator,
+    EC2TerminateInstanceOperator,
+)
 
 
 class BaseEc2TestClass:
     @classmethod
-    def _create_instance(cls, hook: EC2Hook):
-        """Create Instance and return instance id."""
+    def _get_image_id(cls, hook):
+        """Get a valid image id to create an instance."""
         conn = hook.get_conn()
         try:
             ec2_client = conn.meta.client
         except AttributeError:
             ec2_client = conn
-
-        # We need existed AMI Image ID otherwise `moto` will raise DeprecationWarning.
+        
+        # We need an existing AMI Image ID otherwise `moto` will raise DeprecationWarning.
         images = ec2_client.describe_images()["Images"]
-        response = ec2_client.run_instances(MaxCount=1, MinCount=1, ImageId=images[0]["ImageId"])
-        return response["Instances"][0]["InstanceId"]
+        return images[0]["ImageId"]
+
+class TestEC2CreateInstanceOperator(BaseEc2TestClass):
+    def test_init(self):
+        ec2_operator = EC2CreateInstanceOperator(
+            task_id="test_create_instance",
+            image_id="test_image_id",
+        )
+
+        assert ec2_operator.task_id == "test_create_instance"
+        assert ec2_operator.image_id == "test_image_id"
+        assert ec2_operator.max_count == 1
+        assert ec2_operator.min_count == 1
+        assert ec2_operator.max_attempts == 20
+        assert ec2_operator.poll_interval == 20
+
+    @mock_ec2
+    def test_create_intance(self):
+        ec2_hook = EC2Hook()
+        create_instance = EC2CreateInstanceOperator(
+            image_id=self._get_image_id(ec2_hook),
+            task_id="test_create_instance",
+        )
+        instance_id = create_instance.execute(None)
+
+        assert ec2_hook.get_instance_state(instance_id=instance_id) == "running"
+
+
+class TestEC2TerminateInstanceOperator(BaseEc2TestClass):
+    def test_init(self):
+        ec2_operator = EC2TerminateInstanceOperator(
+            task_id="test_terminate_instance",
+            instance_id="test_image_id",
+        )
+
+        assert ec2_operator.task_id == "test_terminate_instance"
+        assert ec2_operator.max_attempts == 20
+        assert ec2_operator.poll_interval == 20
+
+    @mock_ec2
+    def test_terminate_intance(self):
+        ec2_hook = EC2Hook()
+
+        create_instance = EC2CreateInstanceOperator(
+            image_id=self._get_image_id(ec2_hook),
+            task_id="test_create_instance",
+        )
+        instance_id = create_instance.execute(None)
+
+        assert ec2_hook.get_instance_state(instance_id=instance_id) == "running"
+
+        terminate_instance = EC2TerminateInstanceOperator(
+            task_id="test_terminate_instance", instance_id=instance_id
+        )
+        terminate_instance.execute(None)
+
+        assert ec2_hook.get_instance_state(instance_id=instance_id) == "terminated"
 
 
 class TestEC2StartInstanceOperator(BaseEc2TestClass):
@@ -58,7 +118,11 @@ class TestEC2StartInstanceOperator(BaseEc2TestClass):
     def test_start_instance(self):
         # create instance
         ec2_hook = EC2Hook()
-        instance_id = self._create_instance(ec2_hook)
+        create_instance = EC2CreateInstanceOperator(
+            image_id=self._get_image_id(ec2_hook),
+            task_id="test_create_instance",
+        )
+        instance_id = create_instance.execute(None)
 
         # start instance
         start_test = EC2StartInstanceOperator(
@@ -89,7 +153,11 @@ class TestEC2StopInstanceOperator(BaseEc2TestClass):
     def test_stop_instance(self):
         # create instance
         ec2_hook = EC2Hook()
-        instance_id = self._create_instance(ec2_hook)
+        create_instance = EC2CreateInstanceOperator(
+            image_id=self._get_image_id(ec2_hook),
+            task_id="test_create_instance",
+        )
+        instance_id = create_instance.execute(None)
 
         # stop instance
         stop_test = EC2StopInstanceOperator(
@@ -99,3 +167,4 @@ class TestEC2StopInstanceOperator(BaseEc2TestClass):
         stop_test.execute(None)
         # assert instance state is running
         assert ec2_hook.get_instance_state(instance_id=instance_id) == "stopped"
+

--- a/tests/providers/amazon/aws/operators/test_ec2.py
+++ b/tests/providers/amazon/aws/operators/test_ec2.py
@@ -37,10 +37,11 @@ class BaseEc2TestClass:
             ec2_client = conn.meta.client
         except AttributeError:
             ec2_client = conn
-        
+
         # We need an existing AMI Image ID otherwise `moto` will raise DeprecationWarning.
         images = ec2_client.describe_images()["Images"]
         return images[0]["ImageId"]
+
 
 class TestEC2CreateInstanceOperator(BaseEc2TestClass):
     def test_init(self):
@@ -65,14 +66,14 @@ class TestEC2CreateInstanceOperator(BaseEc2TestClass):
         )
         instance_id = create_instance.execute(None)
 
-        assert ec2_hook.get_instance_state(instance_id=instance_id) == "running"
+        assert ec2_hook.get_instance_state(instance_id=instance_id[0]) == "running"
 
 
 class TestEC2TerminateInstanceOperator(BaseEc2TestClass):
     def test_init(self):
         ec2_operator = EC2TerminateInstanceOperator(
             task_id="test_terminate_instance",
-            instance_id="test_image_id",
+            instance_ids="test_image_id",
         )
 
         assert ec2_operator.task_id == "test_terminate_instance"
@@ -89,14 +90,14 @@ class TestEC2TerminateInstanceOperator(BaseEc2TestClass):
         )
         instance_id = create_instance.execute(None)
 
-        assert ec2_hook.get_instance_state(instance_id=instance_id) == "running"
+        assert ec2_hook.get_instance_state(instance_id=instance_id[0]) == "running"
 
         terminate_instance = EC2TerminateInstanceOperator(
-            task_id="test_terminate_instance", instance_id=instance_id
+            task_id="test_terminate_instance", instance_ids=instance_id
         )
         terminate_instance.execute(None)
 
-        assert ec2_hook.get_instance_state(instance_id=instance_id) == "terminated"
+        assert ec2_hook.get_instance_state(instance_id=instance_id[0]) == "terminated"
 
 
 class TestEC2StartInstanceOperator(BaseEc2TestClass):
@@ -127,11 +128,11 @@ class TestEC2StartInstanceOperator(BaseEc2TestClass):
         # start instance
         start_test = EC2StartInstanceOperator(
             task_id="start_test",
-            instance_id=instance_id,
+            instance_id=instance_id[0],
         )
         start_test.execute(None)
         # assert instance state is running
-        assert ec2_hook.get_instance_state(instance_id=instance_id) == "running"
+        assert ec2_hook.get_instance_state(instance_id=instance_id[0]) == "running"
 
 
 class TestEC2StopInstanceOperator(BaseEc2TestClass):
@@ -162,9 +163,8 @@ class TestEC2StopInstanceOperator(BaseEc2TestClass):
         # stop instance
         stop_test = EC2StopInstanceOperator(
             task_id="stop_test",
-            instance_id=instance_id,
+            instance_id=instance_id[0],
         )
         stop_test.execute(None)
         # assert instance state is running
-        assert ec2_hook.get_instance_state(instance_id=instance_id) == "stopped"
-
+        assert ec2_hook.get_instance_state(instance_id=instance_id[0]) == "stopped"

--- a/tests/system/providers/amazon/aws/example_ec2.py
+++ b/tests/system/providers/amazon/aws/example_ec2.py
@@ -24,7 +24,12 @@ import boto3
 from airflow import DAG
 from airflow.decorators import task
 from airflow.models.baseoperator import chain
-from airflow.providers.amazon.aws.operators.ec2 import EC2StartInstanceOperator, EC2StopInstanceOperator
+from airflow.providers.amazon.aws.operators.ec2 import (
+    EC2CreateInstanceOperator,
+    EC2StartInstanceOperator,
+    EC2StopInstanceOperator,
+    EC2TerminateInstanceOperator,
+)
 from airflow.providers.amazon.aws.sensors.ec2 import EC2InstanceStateSensor
 from airflow.utils.trigger_rule import TriggerRule
 from tests.system.providers.amazon.aws.utils import ENV_ID_KEY, SystemTestContextBuilder
@@ -62,35 +67,6 @@ def create_key_pair(key_name: str):
     return key_pair_id
 
 
-@task
-def create_instance(instance_name: str, key_pair_id: str):
-    client = boto3.client("ec2")
-
-    # Create the instance
-    instance_id = client.run_instances(
-        ImageId=_get_latest_ami_id(),
-        MinCount=1,
-        MaxCount=1,
-        InstanceType="t2.micro",
-        KeyName=key_pair_id,
-        TagSpecifications=[{"ResourceType": "instance", "Tags": [{"Key": "Name", "Value": instance_name}]}],
-        # Use IMDSv2 for greater security, see the following doc for more details:
-        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
-        MetadataOptions={"HttpEndpoint": "enabled", "HttpTokens": "required"},
-    )["Instances"][0]["InstanceId"]
-
-    # Wait for it to exist
-    waiter = client.get_waiter("instance_status_ok")
-    waiter.wait(InstanceIds=[instance_id])
-
-    return instance_id
-
-
-@task(trigger_rule=TriggerRule.ALL_DONE)
-def terminate_instance(instance: str):
-    boto3.client("ec2").terminate_instances(InstanceIds=[instance])
-
-
 @task(trigger_rule=TriggerRule.ALL_DONE)
 def delete_key_pair(key_pair_id: str):
     boto3.client("ec2").delete_key_pair(KeyName=key_pair_id)
@@ -105,43 +81,78 @@ with DAG(
 ) as dag:
     test_context = sys_test_context_task()
     env_id = test_context[ENV_ID_KEY]
-
+    instance_name = f"{env_id}-instance"
     key_name = create_key_pair(key_name=f"{env_id}_key_pair")
-    instance_id = create_instance(instance_name=f"{env_id}-instance", key_pair_id=key_name)
+
+    config = {
+        "InstanceType": "t2.micro",
+        "KeyName": key_name,
+        "TagSpecifications": [
+            {"ResourceType": "instance", "Tags": [{"Key": "Name", "Value": instance_name}]}
+        ],
+        # Use IMDSv2 for greater security, see the following doc for more details:
+        # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-service.html
+        "MetadataOptions": {"HttpEndpoint": "enabled", "HttpTokens": "required"},
+    }
+    # [START howto_operator_ec2_create_instance]
+    create_instance = EC2CreateInstanceOperator(
+        task_id="create_instance",
+        image_id=_get_latest_ami_id(),
+        max_count=1,
+        min_count=1,
+        config=config,
+        wait_for_completion=True,
+    )
+    # [END howto_operator_ec2_create_instance]
+
+    # [START howto_operator_ec2_stop_instance]
+    stop_instance_1 = EC2StopInstanceOperator(
+        task_id="stop_instance_1",
+        instance_id=create_instance.output,
+    )
+    # [END howto_operator_ec2_stop_instance]
+    stop_instance_1.trigger_rule = TriggerRule.ALL_DONE
+
     # [START howto_operator_ec2_start_instance]
     start_instance = EC2StartInstanceOperator(
         task_id="start_instance",
-        instance_id=instance_id,
+        instance_id=create_instance.output,
     )
     # [END howto_operator_ec2_start_instance]
 
     # [START howto_sensor_ec2_instance_state]
     await_instance = EC2InstanceStateSensor(
         task_id="await_instance",
-        instance_id=instance_id,
+        instance_id=create_instance.output,
         target_state="running",
     )
     # [END howto_sensor_ec2_instance_state]
 
-    # [START howto_operator_ec2_stop_instance]
-    stop_instance = EC2StopInstanceOperator(
-        task_id="stop_instance",
-        instance_id=instance_id,
+    stop_instance_2 = EC2StopInstanceOperator(
+        task_id="stop_instance_2",
+        instance_id=create_instance.output,
+        trigger_rule=TriggerRule.ALL_DONE,
     )
-    # [END howto_operator_ec2_stop_instance]
-    stop_instance.trigger_rule = TriggerRule.ALL_DONE
-
+    # [START howto_operator_ec2_terminate_instance]
+    terminate_instance = EC2TerminateInstanceOperator(
+        task_id="terminate_instance",
+        instance_id=create_instance.output,
+        wait_for_completion=True,
+    )
+    # [END howto_operator_ec2_terminate_instance]
+    terminate_instance.trigger_rule = TriggerRule.ALL_DONE
     chain(
         # TEST SETUP
         test_context,
         key_name,
-        instance_id,
         # TEST BODY
+        create_instance,
+        stop_instance_1,
         start_instance,
         await_instance,
-        stop_instance,
+        stop_instance_2,
+        terminate_instance,
         # TEST TEARDOWN
-        terminate_instance(instance_id),
         delete_key_pair(key_name),
     )
 

--- a/tests/system/providers/amazon/aws/example_ec2.py
+++ b/tests/system/providers/amazon/aws/example_ec2.py
@@ -39,7 +39,8 @@ DAG_ID = "example_ec2"
 sys_test_context_task = SystemTestContextBuilder().build()
 
 
-def _get_latest_ami_id():
+@task
+def get_latest_ami_id():
     """Returns the AMI ID of the most recently-created Amazon Linux image"""
 
     # Amazon is retiring AL2 in 2023 and replacing it with Amazon Linux 2022.
@@ -88,7 +89,7 @@ with DAG(
     env_id = test_context[ENV_ID_KEY]
     instance_name = f"{env_id}-instance"
     key_name = create_key_pair(key_name=f"{env_id}_key_pair")
-    image_id = _get_latest_ami_id()
+    image_id = get_latest_ami_id()
 
     config = {
         "InstanceType": "t2.micro",
@@ -150,6 +151,7 @@ with DAG(
         # TEST SETUP
         test_context,
         key_name,
+        image_id,
         # TEST BODY
         create_instance,
         instance_id,

--- a/tests/system/providers/amazon/aws/example_ec2.py
+++ b/tests/system/providers/amazon/aws/example_ec2.py
@@ -72,6 +72,11 @@ def delete_key_pair(key_pair_id: str):
     boto3.client("ec2").delete_key_pair(KeyName=key_pair_id)
 
 
+@task
+def parse_response(instance_ids: list):
+    return instance_ids[0]
+
+
 with DAG(
     dag_id=DAG_ID,
     schedule="@once",
@@ -109,11 +114,11 @@ with DAG(
     )
     # [END howto_operator_ec2_create_instance]
     create_instance.wait_for_completion = True
-
+    instance_id = parse_response(create_instance.output)
     # [START howto_operator_ec2_stop_instance]
     stop_instance = EC2StopInstanceOperator(
         task_id="stop_instance",
-        instance_id=create_instance.output,
+        instance_id=instance_id,
     )
     # [END howto_operator_ec2_stop_instance]
     stop_instance.trigger_rule = TriggerRule.ALL_DONE
@@ -121,14 +126,14 @@ with DAG(
     # [START howto_operator_ec2_start_instance]
     start_instance = EC2StartInstanceOperator(
         task_id="start_instance",
-        instance_id=create_instance.output,
+        instance_id=instance_id,
     )
     # [END howto_operator_ec2_start_instance]
 
     # [START howto_sensor_ec2_instance_state]
     await_instance = EC2InstanceStateSensor(
         task_id="await_instance",
-        instance_id=create_instance.output,
+        instance_id=instance_id,
         target_state="running",
     )
     # [END howto_sensor_ec2_instance_state]
@@ -136,7 +141,7 @@ with DAG(
     # [START howto_operator_ec2_terminate_instance]
     terminate_instance = EC2TerminateInstanceOperator(
         task_id="terminate_instance",
-        instance_id=create_instance.output,
+        instance_ids=instance_id,
         wait_for_completion=True,
     )
     # [END howto_operator_ec2_terminate_instance]
@@ -147,6 +152,7 @@ with DAG(
         key_name,
         # TEST BODY
         create_instance,
+        instance_id,
         stop_instance,
         start_instance,
         await_instance,

